### ISSUE TITLE
Crash when trying to build cex for an array of floats

### ISSUE THREAD

**Issue by mikhailramalho** [esbmc/esbmc#287]
Reduced case for boolector, yices, mathsat:
```
float a[][6];
int b;
float c;
main() {
  c = a[b][5];
  assert(c);
}
```

**Comment by mikhailramalho** [comment]
Reduced case for Z3:
```
a[][3];
b;
main() {
  int c, d = a[c][2];
  __VERIFIER_assume(d >= 0);
  assert(b);
}
```

**Comment by mikhailramalho** [comment]
It turns out the CVC4 issue is not related to arrays, but to floating-point encoding. I created another issue for it, issue #288.

**Comment by mikhailramalho** [comment]
Fixes for Boolector, mathsat and yices are available in branch `fix-float-array-cex`.

**Comment by mikhailramalho** [comment]
fixed by PR #289
